src/pbft/config/config.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright (c) 2018 TrueChain Foundation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"log"
+	"os"
+)
+
+const (
+	configFile               = "/.truechain_config.yml"
+	trueChainConsensusConfig = "TRUECHAIN_CONSENSUS_CONFIGURATION"
+)
+
+// Config  configuration for pbft
+type Config struct {
+	// general
+	BasePort          int   `yaml:"basePort"`
```

Comment:
Since we are using separate ports for getting grpc requests for new transactions from clients, we would probably need to have a GrpcBasePort member here too.

In general this `Config` struct declaration needs to be in sync with the `Config` struct as declared inside `src/pbft-core/config.go`. Hence instead of BasePort, we would need a list of ports, and a separate list of grpc ports.

Reply:
I noticed that these changes are for the `devel` branch. So the above comments might not apply which are more relevant to the `master` bracnh.

Reply:
Not exactly get this. Will talk to archit about this branch.